src/transformers/models/oneformer/processing_oneformer.py

```diff
@@ -48,22 +69,23 @@ class OneFormerProcessor(ProcessorMixin):
     tokenizer_class = ("CLIPTokenizer", "CLIPTokenizerFast")
 
     def __init__(
-        self, image_processor=None, tokenizer=None, max_seq_length: int = 77, task_seq_length: int = 77, **kwargs
+        self,
+        image_processor=None,
+        tokenizer=None,
+        max_seq_length: int = 77,
+        task_seq_length: int = 77,
```

Comment:
so the way it should be is that in `OneFormerProcessorKwargs`, these defaults should be passed to the relevant dictionary, which you can then get back. say
```python
_defaults = {"max_seq_length" = 77,
        "task_seq_length" = 77}
```

and you can inform the types with a Kwargs typed class

```python
class OneFormerTextKwargs(TextKwargs, total=False):
    max_seq_length: Optional[int]
    task_seq_length: Optional[int]
```
then your types are correctly informed as well as your defaults and you can use both.

Reply:
This was a bigger refactor. Let me know what you think...